pandas/core/arrays/datetimelike.py

```diff
@@ -1380,25 +1381,15 @@ def _addsub_object_array(self, other: np.ndarray, op):
         return result
 
     def _accumulate(self, name: str, *, skipna: bool = True, **kwargs):
+        if name not in {"cummin", "cummax"}:
+            raise TypeError(f"Accumulation {name} not supported for {type(self)}")
 
-        if is_period_dtype(self.dtype):
-            data = self
-        else:
-            # Incompatible types in assignment (expression has type
-            # "ndarray[Any, Any]", variable has type "DatetimeLikeArrayMixin"
-            data = self._ndarray.copy()  # type: ignore[assignment]
-
-        if name in {"cummin", "cummax"}:
-            func = np.minimum.accumulate if name == "cummin" else np.maximum.accumulate
-            result = cast(np.ndarray, nanops.na_accum_func(data, func, skipna=skipna))
-
-            # error: Unexpected keyword argument "freq" for
-            # "_simple_new" of "NDArrayBacked"  [call-arg]
-            return type(self)._simple_new(
-                result, freq=self.freq, dtype=self.dtype  # type: ignore[call-arg]
-            )
+        op = getattr(datetimelike_accumulations, name)
+        result = op(self.copy(), skipna=skipna, **kwargs)
 
-        raise TypeError(f"Accumulation {name} not supported for {type(self)}")
+        return type(self)._simple_new(
+            result, freq=self.freq, dtype=self.dtype  # type: ignore[call-arg]
```

Comment:
i dont think retaining self.freq is going to be correct in general

Reply:
Can you elaborate when we should set freq to None?

Reply:
Pretty much anything that isn't a no-op should be not-freq-preserving.  e.g. `pd.date_range("2016-01-01", periods=3)._data.cummin()`

Reply:
Thx, makes sense. Should we retain the freq in cases where we only have one element or not retain in general?

Not easy to hit btw :) But added a test now

Reply:
id just not-retain in general

Reply:
Sounds good, that's how I implemented it right now